app/controllers/dashboard_controller.rb

```diff
@@ -689,8 +690,9 @@ def get_session_data
   end
 
   def identity_provider_login(identity_type)
-    if @user_name.blank? && request.env.key?("HTTP_X_REMOTE_USER").present?
-      @user_name = params[:user_name] = request.env["HTTP_X_REMOTE_USER"].split("@").first
+    if @user_name.blank? && request.env["HTTP_X_REMOTE_USER"].present?
+      user_env = request.env["HTTP_X_REMOTE_USER"].force_encoding("UTF-8")
```

Comment:
why is this named `user_env` rather than `user_header` as above?

Reply:
I think it's `user_header` above because it's the user pulled from `request.headers`.
It's `user_env` here because it's the user pulled from `request.env`.